src/proxy/momento/src/frontend.rs

```diff
@@ -80,132 +81,129 @@ pub(crate) async fn handle_resp_client(
+                }
+                resp::Request::HashGet(r) => {
+                    resp::hget(&mut client, &cache_name, &mut socket, r.key(), r.field()).await?
+                }
+                resp::Request::HashGetAll(r) => {
+                    resp::hgetall(&mut client, &cache_name, &mut socket, r.key()).await?
+                }
+                resp::Request::HashIncrBy(r) => {
+                    resp::hincrby(&mut client, &cache_name, &mut socket, r).await?
+                }
+                resp::Request::HashKeys(r) => {
+                    resp::hkeys(&mut client, &cache_name, &mut socket, r.key()).await?
+                }
+                resp::Request::HashLength(r) => {
+                    resp::hlen(&mut client, &cache_name, &mut socket, r.key()).await?
+                }
+                resp::Request::HashMultiGet(r) => {
+                    resp::hmget(&mut client, &cache_name, &mut socket, r.key(), r.fields()).await?
+                }
+                resp::Request::HashSet(r) => {
+                    resp::hset(&mut client, &cache_name, &mut socket, r.key(), r.data()).await?
+                }
+                resp::Request::HashValues(r) => {
+                    resp::hvals(&mut client, &cache_name, &mut socket, r.key()).await?
+                }
+                resp::Request::Set(r) => {
+                    resp::set(&mut client, &cache_name, &mut socket, &r).await?
+                }
+                resp::Request::SetAdd(r) => {
+                    resp::sadd(&mut client, &cache_name, &mut socket, &r).await?
+                }
+                _ => return Err(ProxyError::UnsupportedCommand),
+            }
+
+            Ok(())
+        }
+        .await;
+
+        let fatal = match result {
+            Ok(()) => false,
+            Err(e) => {
+                response_buf.clear();
+
+                match e {
+                    ProxyError::Momento(error) => {
+                        SESSION_SEND.increment();
+                        crate::protocol::resp::momento_error_to_resp_error(
+                            &mut response_buf,
+                            command,
+                            error,
+                        );
+
+                        false
+                    }
+                    ProxyError::Timeout(_) => {
+                        SESSION_SEND.increment();
+                        BACKEND_EX.increment();
+                        BACKEND_EX_TIMEOUT.increment();
+                        response_buf.extend_from_slice(b"-ERR backend timeout\r\n");
+
+                        false
+                    }
+                    ProxyError::Io(_) => true,
+                    ProxyError::UnsupportedCommand => {
+                        println!("bad request");
+                        response_buf.extend_from_slice(b"CLIENT_ERROR\r\n");
+                        true
+                    }
+                }
+            }
+        };
+
+        // Temporary workaround
```

Comment:
Seems like this should generally work. I guess memcache commands with noreply set could be an issue... but we probably don't handle that properly right now anyway.

Reply:
Yeah this is only for the resp command set. I didn't check that it would work anywhere else and plan to remove it once I've had time to go through and convert all the commands anyway.